src/python/foglamp/core/server.py

```diff
@@ -40,15 +55,118 @@ def _make_app():
+
+        :rtype: web.Application
+        """
+        app = web.Application(middlewares=[middleware.error_middleware])
+        routes.core_setup(app)
+        return app
+
     @classmethod
     async def _start_scheduler(cls):
         """Starts the scheduler"""
-        cls.scheduler = Scheduler()
+        _logger.info("start scheduler")
+        cls.scheduler = Scheduler(cls.core_management_port)
         await cls.scheduler.start()
 
+    @staticmethod
+    def __start_storage(host, m_port):
+        _logger.info("start storage, from directory %s", _STORAGE_DIR)
+        try:
+            cmd_with_args = ['./storage', '--address={}'.format(host),
+                             '--port={}'.format(m_port)]
+            subprocess.call(cmd_with_args, cwd=_STORAGE_DIR)
+        except Exception as ex:
+            _logger.exception(str(ex))
+
+    @classmethod
+    async def _start_storage(cls, loop):
+        if loop is None:
+            loop = asyncio.get_event_loop()
+            # callback with args
+        loop.call_soon(cls.__start_storage, cls._host, cls.core_management_port)
+
+    @classmethod
+    def _start_app(cls, loop, app, host, port):
+        if loop is None:
+            loop = asyncio.get_event_loop()
+
+        handler = app.make_handler()
+        coro = loop.create_server(handler, host, port)
+        # added coroutine
+        server = loop.run_until_complete(coro)
+        return server, handler
+
+    @classmethod
+    def _start_core(cls, loop=None):
+        _logger.info("start core")
+
+        try:
+            host = cls._host
+
+            core_app = cls._make_core_app()
+            core_server, core_server_handler = cls._start_app(loop, core_app, host, 0)
+            address, cls.core_management_port = core_server.sockets[0].getsockname()
+            _logger.info('Management API started on http://%s:%s', address, cls.core_management_port)
+
+            # start storage
+            loop.run_until_complete(cls._start_storage(loop))
+            # start scheduler
+            # scheduler on start will wait for storage service registration
+            loop.run_until_complete(cls._start_scheduler())
+
+            service_app = cls._make_app()
+            service_server, service_server_handler = cls._start_app(loop, service_app, host, cls.rest_service_port)
+            address, service_server_port = service_server.sockets[0].getsockname()
+            _logger.info('Rest Server started on http://%s:%s', address, service_server_port)
+
+            # register core
+            # a service with 2 web server instance,
+            # registering now only when service_port is ready to listen the request
+            cls._register_core(host, cls.core_management_port, cls.rest_service_port)
+            print("(Press CTRL+C to quit)")
+            try:
+                loop.run_forever()
+            except KeyboardInterrupt:
+                pass
+            finally:
+                cls.stop_storage()
```

Comment:
Per [shutdown sequence](https://docs.google.com/document/d/1MTFrwyXIa7YEPAIpnlNmDG7X1qFie9SL8PW2YT-hOkU/edit#heading=h.s8nqjramfg0a), storage service shutdown should come at last.

Reply:
@amarendra-dianomic  Yes that is  to make sure to unregister it. Also here we have redundancy too signal call binding + call in daemon. 

* see foglamp.core.server.Server#_stop_core (stub comments)
Also refer to my comment  https://github.com/foglamp/FogLAMP/pull/190#issuecomment-336679199
* service shutdown / ping
* kill signal binding fix and shutdown/ clean up of aiohttp -- Hence a single method; _stop with three internal calls _stop_scheduler, _stop_storage, _stop_core

Let's handle these things in separate tickets, post merge of this PR. 

Also Device Service.Type is now changed to `Southbound`|| `SouthBound`